### PR TITLE
Update for NCS v2.4.x

### PR DESCRIPTION
**Pull request by hieuv**

### CODE REVIEW

File: bootloader_samples/smp/mcuboot_smp_uart/prj.conf [hunk 1]
```diff
@@ -1,4 +1,21 @@
 CONFIG_BOOTLOADER_MCUBOOT=y
 CONFIG_MCUMGR=y
-CONFIG_MCUMGR_SMP_UART=y
-CONFIG_MCUMGR_CMD_IMG_MGMT=y
+CONFIG_MCUMGR_TRANSPORT_UART=y
+CONFIG_MCUMGR_GRP_IMG=y
+
+# CONFIG_MCUMGR_TRANSPORT_UART dependencies
```
Comment (hieuv) [inline bootloader_samples/smp/mcuboot_smp_uart/prj.conf:6]: These used to be selected by CONFIG_MCUMGR_SMP_UART.
	https://github.com/nrfconnect/sdk-zephyr/blob/v3.2.99-ncs2/subsys/mgmt/mcumgr/transport/Kconfig.uartL5-L14
Now they are not selected by, only depended on by CONFIG_MCUMGR_TRANSPORT_UART
	https://github.com/nrfconnect/sdk-zephyr/blob/v3.3.99-ncs1/subsys/mgmt/mcumgr/transport/Kconfig.uartL12-L21

File: bootloader_samples/smp/mcuboot_smp_uart/prj.conf [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+CONFIG_BASE64=y
+
+# CONFIG_MCUMGR_GRP_IMG dependencies
```
Comment (hieuv) [inline bootloader_samples/smp/mcuboot_smp_uart/prj.conf:9]: These used to be selected by CONFIG_MCUMGR_CMD_IMG_MGMT.
https://github.com/nrfconnect/sdk-zephyr/blob/v3.2.99-ncs2/subsys/mgmt/mcumgr/grp/img_mgmt/Kconfig L5-L12
Now they are not selected by, only depended on by CONFIG_MCUMGR_GRP_IMG
https://github.com/nrfconnect/sdk-zephyr/blob/v3.3.99-ncs1/subsys/mgmt/mcumgr/grp/img_mgmt/Kconfig L15-L23

File: bootloader_samples/smp/mcuboot_smp_uart/prj.conf [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+
+# CONFIG_IMG_MANAGER dependencies
+CONFIG_STREAM_FLASH=y
+
+# CONFIG_MCUMGR dependencies
+CONFIG_NET_BUF=y
+CONFIG_ZCBOR=y
+
+# Required for CONFIG_IMG_MANAGER
+CONFIG_FLASH_MAP=y
```
Comment (hieuv) [inline bootloader_samples/smp/mcuboot_smp_uart/prj.conf:21]: Same "selected by" becomes "depended on by" situation.
https://github.com/nrfconnect/sdk-zephyr/commit/c3bfe7a6baf7e47d8ad70e38ffdea313f3ec62ce

**Review by hellesvik-nordic — APPROVED** [review]
Changes looks good. Also tested and it swaps fine

**Pull request merged**